core/src/main/scala/kafka/server/ReplicaManager.scala

```diff
@@ -898,6 +898,8 @@ class ReplicaManager(val config: KafkaConfig,
                    Errors.COORDINATOR_NOT_AVAILABLE |
                    Errors.NOT_COORDINATOR => Some(new NotEnoughReplicasException(
                 s"Unable to verify the partition has been added to the transaction. Underlying error: ${error.toString}"))
+              case Errors.UNKNOWN_PRODUCER_ID => Some(new OutOfOrderSequenceException(
```

Comment:
Do we actually return `UNKNOWN_PRODUCER_ID`?

Reply:
You are right. UNKNOWN_PRODUCER_ID is not returned anywhere.